source/common/common/thread.cc

```diff
@@ -21,7 +25,15 @@ Thread::Thread(std::function<void()> thread_routine) : thread_routine_(thread_ro
   UNREFERENCED_PARAMETER(rc);
 }
 
-int32_t Thread::currentThreadId() { return syscall(SYS_gettid); }
+int32_t Thread::currentThreadId() {
+#ifdef __linux__
```

Comment:
Nit: could you either add `#else` case or use Linux as the fallback? i.e.

    #if defined(__linux__)
      return syscall(SYS_gettid);
    #elif defined(__APPLE__)
      uint64_t tid;
      pthread_threadid_np(NULL, &tid);
      return tid;
    #else
    #error "OS not supported."
    #endif

or

    #ifdef __APPLE__
      uint64_t tid;
      pthread_threadid_np(NULL, &tid);
      return tid;
    #else
      return syscall(SYS_gettid);
    #endif

otherwise we're going to end with code that doesn't have return value on other OSes.

Either one is fine with me, assuming @mattklein123 doesn't have any objections.